actix-http/src/body/body.rs

```diff
@@ -33,29 +34,60 @@ impl AnyBody {
         Self::Bytes(Bytes::new())
     }
 
-    /// Create body from slice (copy)
+    /// Create boxed body from generic message body.
+    pub fn new_boxed<B>(body: B) -> Self
+    where
+        B: MessageBody + 'static,
+        B::Error: Into<Box<dyn StdError + 'static>>,
+    {
+        Self::Body(BoxBody::from_body(body))
+    }
+
+    /// Constructs new `AnyBody` instance from a slice of bytes by copying it.
+    ///
+    /// If your bytes container is owned, it may be cheaper to use a `From` impl.
+    pub fn copy_from_slice(s: &[u8]) -> Self {
+        Self::Bytes(Bytes::copy_from_slice(s))
+    }
+
+    #[doc(hidden)]
+    #[deprecated(since = "4.0.0", note = "Renamed to `copy_from_slice`.")]
     pub fn from_slice(s: &[u8]) -> Self {
         Self::Bytes(Bytes::copy_from_slice(s))
     }
+}
 
+impl<B> AnyBody<B>
+where
+    B: MessageBody + 'static,
+    B::Error: Into<Box<dyn StdError + 'static>>,
+{
     /// Create body from generic message body.
-    pub fn from_message<B>(body: B) -> Self
-    where
-        B: MessageBody + 'static,
-        B::Error: Into<Box<dyn StdError + 'static>>,
-    {
-        Self::Stream(BoxAnyBody::from_body(body))
+    pub fn new(body: B) -> Self {
+        Self::Body(body)
+    }
+
+    pub fn into_boxed(self) -> AnyBody {
+        match self {
+            AnyBody::None => AnyBody::new_boxed(()),
+            AnyBody::Bytes(body) => AnyBody::new_boxed(body),
+            AnyBody::Body(body) => AnyBody::new_boxed(body),
+        }
     }
 }
 
-impl MessageBody for AnyBody {
+impl<B> MessageBody for AnyBody<B>
+where
+    B: MessageBody + Unpin,
```

Comment:
This does not seems right to me. The AnyBody<B> API should serve the purpose of pinning the B type and only box it when necessary. In general I think the response body should be a type with pinned B type and the box of it happens should at the top level where body type erase is necessary(or a move must happen)

Reply:
An example:
```rust
pin_project_lite::pin_project! {
    struct Body<B = futures_util::stream::BoxStream<'static, ()>> {
        #[pin]
        body: B
    }
}

fn unpin<T: Unpin>(_t: &T) {}

fn test() {
    let body = Body {
        body: Box::pin(futures_util::stream::once(async {}))
    };

    unpin(&body); // infered correctly.

    let body = Body {
        body: futures_util::stream::once(async {})
    };

    unpin(&body); // compile error
}
```

Reply:
well spotted, the bound has been removed